call-home/src/common/iterator.rs

```diff
@@ -0,0 +1,69 @@
+#[derive(Debug, Clone)]
+pub enum SortOrder {
+    NonDescending,
+    NonAscending,
+}
+
+#[derive(Debug, Clone)]
+pub struct SortedExactSizeIter<T> {
+    data: Vec<T>,
+    sort_order: SortOrder,
+}
+
+impl<T: Ord> SortedExactSizeIter<T> {
```

Comment:
Why do we need this?

Reply:
This type guarantees that it's sorted. That takes makes max, min and reduces the serveral sorts in `get_percentile`, while making sure its input is sorted with this type.

Reply:
Why can't we just sort it inline? It's trivial and would avoid adding this whole file.